core/src/main/java/org/infinispan/interceptors/impl/CacheWriterInterceptor.java

```diff
@@ -461,14 +463,16 @@ public int getNumberOfPersistedEntries() {
 
    void storeEntry(InvocationContext ctx, Object key, FlagAffectedCommand command) {
       MarshalledEntry entry = createMarshalledEntry(ctx, key);
-      persistenceManager.writeToAllNonTxStores(entry, SegmentSpecificCommand.extractSegment(command, key, keyPartitioner),
-            skipSharedStores(ctx, key, command) ? PRIVATE : BOTH, command.getFlagsBitSet());
-      if (trace) getLog().tracef("Stored entry %s under key %s", entry.getValue(), key);
+      if (entry != null) {
+         persistenceManager.writeToAllNonTxStores(entry, SegmentSpecificCommand.extractSegment(command, key, keyPartitioner),
+               skipSharedStores(ctx, key, command) ? PRIVATE : BOTH, command.getFlagsBitSet());
+         if (trace) getLog().tracef("Stored entry %s under key %s", entry.getValue(), key);
+      }
    }
 
    MarshalledEntry createMarshalledEntry(InvocationContext ctx, Object key) {
       InternalCacheValue sv = entryFactory.getValueFromCtxOrCreateNew(key, ctx);
-      return new MarshalledEntryImpl(key, sv.getValue(), internalMetadata(sv), marshaller);
+      return sv != null ? new MarshalledEntryImpl(key, sv.getValue(), internalMetadata(sv), marshaller) : null;
```

Comment:
Again `create` in the method name suggests a non-null return value, I suggest `marshallEntry`

Reply:
+1 with @danberindei the method should never return null

Reply:
Sure I can change the name.